Sources/ChessKit/Board.swift

```diff
@@ -215,8 +215,30 @@ public struct Board {
             delegate?.didEnd(with: .draw(.stalemate))
         } else if position.clock.halfmoves >= Clock.halfMoveMaximum {
             delegate?.didEnd(with: .draw(.fiftyMoves))
-        }
-        
+            
+            // position doesn't contain any pawn/rook/queen -> possible draw
+        } else if !position.pieces.contains(where: {$0.kind == .pawn || $0.kind == .rook || $0.kind == .queen}) {
+            
+                switch position.pieces.count {
+                    
+                    // 0, 1 -> impossible, but still checking
+                    // 2 pieces -> 2 kings = draw
+                    // 3 pieces -> 2 kings + minor piece = draw
+                case 0, 1, 2, 3:
+                    delegate?.didEnd(with: .draw(.insufficientMaterial))
+
+                    // N pieces -> no knight + all bishops on same square color = draw
+                default:
+                    if !position.pieces.contains(where: {$0.kind == .knight}){
+                        let darkSquareBishops = position.pieces.filter({$0.kind != .king && $0.square.color == .dark}).count
+                        
+                        if darkSquareBishops == 0 || darkSquareBishops == (position.pieces.count - 2){
+                            delegate?.didEnd(with: .draw(.insufficientMaterial))
+                        }
+                    }
+                }
+            }
```

Comment:
I would make a few changes here:

#### Use `pieceSet` instead of `pieces`
First, this `process(move:)` function runs every time someone makes a move so to be more performant it might be better to do use the `pieceSet` object in `Position` and use bitwise operators instead of things like `contains(where:)` and `filter()`. This can be tricky if you're not familiar so I can help if you want, otherwise you can check the existing code in `Board.swift` (`canCastle()` for example uses the `pieceSet` and bitwise ops).

#### Move logic to `Position`
Next, to keep this area of the code simpler I would consider moving this logic to `Position` and perhaps having a computed `Bool` property titled something like `hasInsufficientMaterial`. Then in here we can simply check if it's true and if so call `delegate.didEnd(with:)`.

These are just my initial suggestions, happy to hear any other thoughts or ideas.